### ISSUE TITLE
Expert group padding to multiple of 32

### ISSUE THREAD

**Issue by eee4017** [pytorch/torchtitan#2262]
### Bug description

See https://github.com/pytorch/ao/issues/3636

When training MXFP8 MoE models with TorchTitan, expert group sizes must be padded to a multiple of 32 to satisfy MXFP8 scaling-block requirements in the backward pass. Currently, when using quantize via TorchTitan’s model-converter path, this padding does not appear to be consistently enforced, leading to unsupported group sizes (e.g. 44) reaching the MXFP8 CUDA kernels and causing runtime failures or forced fallback to Triton. TorchTitan already has a permutation step that gathers tokens across ranks and can insert padding, so this issue tracks ensuring that expert group padding to a multiple of 32 is always applied before entering the MXFP8 grouped-MM path.

cc @danielvegamyhre 

### Versions

```
pytorch == 2.11.0a0+gita156055
torchtitan == 0.2.1+gita25dd8f
torchao == 0.16.0+gita5f2693
```

```
torchrun \
  --nnodes=2 \
  --nproc-per-node=4 \
  ...
  -m torchtitan.train \
  --job.config_file torchtitan/models/deepseek_v3/train_configs/deepseek_v3_16b.toml \
  --training.steps 50 \
  ...
  --model.converters="quantize.linear.mx,quantize.grouped_mm.mx" \
  --quantize.linear.mx.recipe_name="mxfp8_cublas" \
  --quantize.grouped_mm.mx.fqns="experts" \
  --parallelism.expert_parallel_degree 8
```

**Comment by tianyu-l** [comment]
fyi we are redoing this in https://github.com/pytorch/torchtitan/pull/2255 for a different motivation, so better to solve it altogether

**Comment by danielvegamyhre** [comment]
by the way @eee4017 with regular mxfp8 training, in my single node testing on 8xB200, there is no speedup for the smallest dsv3 model (16b).

However, with MXFP8 expert parallel + `mxfp8_wgrad_with_hp` recipe for the grouped GEMMs (https://github.com/pytorch/torchtitan/pull/2250) I see a 10%+ speedup. So once this lands I would encourage you to try it.

**Comment by danielvegamyhre** [comment]
@eee4017 i think i replied on the torchao issue, but the "44" you're referencing is actually the scale tensor column width (i.e., 1408 dim / 32 mxfp8 scaling factor width = 44), which is correct.

i believe the error you were seeing was due to a cuda kernel for scale factor blocked layout conversion not working properly when the input did not meet 2d TMA alignment constraints. this was solved here (https://github.com/pytorch/ao/pull/3708) and since then i have tested mxfp8 training on dsv3 successful, both in single node and multi node settings, so you should be good to go!

let me know if it is still not working for you or if you encounter other issues
